### ISSUE TITLE
Document structure design

### ISSUE THREAD

**Issue by llxxxll** [sql-machine-learning/sql-machine-learning.github.io#63]
## Document structure design

_- The principle of gradual progress._
_- Users at different stages of learning can find the corresponding guide._
_- Each model should have a corresponding application example._

- Get Started
  - [Installation](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/installation.md)
  - [Running a Demo](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/demo.md)
- SQLFlow Tutorials
  - [The 40s: build TensorFlow DNNClassifier model]( https://www.youtube.com/watch?v=zIkwOQ_davw)
  - [Train model](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/sql_parser.md)
  - Save model
  - Test model
  - Prediction services(Offline)
  - Application scenarios
    - [DNNClassifier example] [The Credit Card Fraud Detection Example](https://github.com/sql-machine-learning/sqlflow/tree/develop/example/creditcard)
    - [DNNRegressor example] 
- Docs
  - Custom Database
    - Parameters
    - Supported Databases
      - the Hive example
      - the MySQL example
  - Data Loading
    - SQL syntax
    - Automatic read columns
  - [Extended SQL Syntax](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/syntax.md)
    - TRAIN
    - WITH
    - COLUMN
    - LABEL
    - PREDICT
    - Extended in Development
  - Commit distributed tasks
  - Prediction services and deployment
- Resources
  - [Models](https://github.com/sql-machine-learning/models)
- Contributions
  - [Build from source code.](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/build.md)
  - [The walkthrough of the source code](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/walkthrough.md)
  - [The choice of parser generator](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/sql_parser.md)
  - [customized model](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/customized%2Bmodel.md)
  - support for other databases
- Features

**Comment by hungry1526** [comment]
**Current structure** 

SQLFlow
 - Installation
 - Design
 - Contribute
GoHive
PySQLFlow

**Proposed structure**

SQLFlow
 - Introduction 
   - What is SQLFlow
   - Motivation 
   - Where can I use SQLFlow
 - Concepts (specific to SQLFlow)
 - Installation
   - Docker Image
   - Build from source
   - Demo/Tutorials
 - More examples 
   - DNNClassifier
   - DNNRegressor
   - etc…
 - Design (Component level details)
   - Submitter, ProducerToConsumer, DB Abstraction layer, Parser, Executer etc…
   - Model
   - GoHive, PySQLFlow, and other connectors
 - Roadmap
 - Team

Team, please review and comment.

**Comment by hungry1526** [comment]
@llxxxll any suggestion?

**Comment by weiguoz** [comment]
Looks good.
What are the `Concepts (specific to SQLFlow)`? would you plz add more details

**Comment by typhoonzero** [comment]
Test model -> model validation
Prediction services(Offline) -> batch predict on a data table

**Comment by hungry1526** [comment]
> Looks good.
> What are the `Concepts (specific to SQLFlow)`? would you plz add more details

This part means adding some definitions for some terminologies we use in SQLFlow, e.g. executer, SQLFlow server, RPC server, etc...For sure the less terms we created, the better.

**Comment by hungry1526** [comment]
> Test model -> model validation
> Prediction services(Offline) -> batch predict on a data table

Thanks for the review, Any suggestion for the structure I proposed?

**Comment by Yancey0623** [comment]
I updated the structure following the latest SQLFlow repo:

* Get Started
    * [QuickStart](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/installation.md), [issues/832](https://github.com/sql-machine-learning/sqlflow/issues/832)
    * [User Guide](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/user_guide.md), [issues/826](https://github.com/sql-machine-learning/sqlflow/issues/826)
    * [run SQLFLow command line](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/demo.md)
* Installation
    * Install SQLFlow using Docker [issue/849](https://github.com/sql-machine-learning/sqlflow/issues/849)
    * Install SQLFlow on Cloud
        * [GCP](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/run_on_gcp.md)
        * AliYun, [issues/836](https://github.com/sql-machine-learning/sqlflow/issues/836)
        * [Kubernetes](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/run_on_kubernetes.md)
    * run SQLFlow with SQL Engines
        * MySQL https://github.com/sql-machine-learning/sqlflow/pull/917
        * Hive via HiveServer2, [issues/835](https://github.com/sql-machine-learning/sqlflow/issues/835)
        * MaxCompute, [issues/834](https://github.com/sql-machine-learning/sqlflow/issues/834)
* Examples
	* [Iris Classification (DNN Classifier)](https://github.com/sql-machine-learning/sqlflow/blob/develop/example/jupyter/tutorial_dnn_iris.ipynb)
	* [Text Classification (LSTM Classifier)](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/text_classification_demo.md)
	* TBD: (Unsupervised ClusterModel )
	* Credit Card Fraud Detection (DNN Classifier), [issues/833](https://github.com/sql-machine-learning/sqlflow/issues/833)
	* [Boston Housing Price (XGBoost Regression Model)]()
	* Analyze Tutorial 
* Models
    * DNNClassifier
    * LSTMClassifier
    * XGBoost Regression Model
    * XGBoost Classification Model
* Design
	* ALPS Design
	* Analyze Design
	* AntXGBoost Design
	* Auth Design
	* ClusterModel Design
	* Customized Model Design
	* ElasticDL on SQLFlow Design
	* Feature Derivation Design
	* ….
* Contributions
	* Build From Source Code
	* Code Walkthrough
	* Custom Model
